libs/vre/shared/app-date-picker/src/lib/app-date-picker/app-date-picker.component.ts

```diff
@@ -509,13 +646,61 @@ export class DatePickerComponent
             month: this.month,
         });
         if (this.disableCalendarSelector) {
-            this.form.controls.calendar.disable();
+            this.form.controls['calendar'].disable();
         } else {
-            this.form.controls.calendar.enable();
+            this.form.controls['calendar'].enable();
         }
         this._setDays(this.calendar, this.era, this.year, this.month);
     }
 
+    /**
+     * given a historical date (year), returns the astronomical year.
+     *
+     * @param year year of the given date.
+     * @param era era of the given date.
+     */
+    convertHistoricalYearToAstronomicalYear(year: number, era: string) {
+        let yearAstro = year;
+        if (era === 'BCE') {
+            // convert historical date to astronomical date
+            yearAstro = yearAstro * -1 + 1;
+        }
+        return yearAstro;
+    }
+
+    /**
+     * calculates the number of days in a month for a given year.
+     *
+     * @param calendar the date's calendar.
+     * @param year the date's year.
+     * @param month the date's month.
+     */
+    calculateDaysInMonth(
+        calendar: string,
+        year: number,
+        month: number
+    ): number {
+        const date = new CalendarDate(year, month, 1);
+        if (calendar === 'GREGORIAN') {
+            const calDate = new GregorianCalendarDate(
+                new CalendarPeriod(date, date)
+            );
+            return calDate.daysInMonth(date);
+        } else if (calendar === 'JULIAN') {
+            const calDate = new JulianCalendarDate(
+                new CalendarPeriod(date, date)
+            );
+            return calDate.daysInMonth(date);
+        } else if (calendar === 'ISLAMIC') {
+            const calDate = new IslamicCalendarDate(
+                new CalendarPeriod(date, date)
+            );
+            return calDate.daysInMonth(date);
+        } else {
+            throw Error('Unknown calendar ' + calendar);
+        }
```

Comment:
Not mandatory but in case there are more than two if/ else if lines you can use switch case

Reply:
I agree but I didn't want to change too much of the code. I'd rather refactor it in a separate PR.